File: StudentPortfolio.WebApp/src/types/dtos/acknowledgement.ts
import { enums } from "../../utilities/enums";
import type { BaseResponse } from "./_baseResponse";
import type { Student } from "./student";

enums.acknowledgementType;

export interface Acknowledgement {
  id: string;
  studentId: number;
  type: keyof typeof enums.acknowledgementType;
  place: string;
  description: string;
  startDate: Date;
  endDate?: Date;
  otherType?: string;
  competitionPosition?: number;
  competitionName?: string;
  studentOrganizatonName?: string;
  student: Student;
}

export type getAcknowledgementResponse = BaseResponse<Acknowledgement>;
export type getAcknowledgementsResponse = BaseResponse<Acknowledgement[]>;

File: StudentPortfolio/Models/Entities/Acknowledgement.cs
using StudentPortfolio.Models.Infrastructure;
using System.ComponentModel.DataAnnotations.Schema;

namespace StudentPortfolio.Models.Entities
{
    public class Acknowledgement : IDeletable, IModel
    {
        public Guid Id { get; set; }

        [Column(TypeName = "tinyint")]
        public AcknowledgementType Type { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string Place { get; set; }
        public string Description { get; set; }
        public DateOnly? StartDate { get; set; }
        public DateOnly? EndDate { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string OtherType { get; set; }
        public bool Deleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }
        public DateTimeOffset DateCreated { get; set; }

        [Column(TypeName = "tinyint")]
        public uint? CompetitionPosition { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string CompetitionName { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string StudentOrganizatonName { get; set; }

        public Guid StudentId { get; set; }
        public Student Student { get; set; }
    }
}
